accounts/backends.py

```diff
@@ -59,9 +59,10 @@ def authenticate(self, request, remote_user, tokens=True):
                 user.is_staff = False
                 user.is_superuser = False
 
-        # Update groups
+        # Update groups: first remove them from existing groups then add them to new groups
+        user.groups.clear()
```

Comment:
We don't want to remove the user from all their groups, just `platform_` ones here (so that way we don't interfere with any groups that our products create)